homeassistant/components/denonavr/media_player.py

```diff
@@ -74,58 +95,98 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
             unique_id = f"{config_entry.unique_id}-{receiver_zone.zone}"
         else:
             unique_id = None
+        await receiver_zone.async_setup()
         entities.append(DenonDevice(receiver_zone, unique_id, config_entry))
     _LOGGER.debug(
         "%s receiver at host %s initialized", receiver.manufacturer, receiver.host
     )
-    async_add_entities(entities)
+    async_add_entities(entities, update_before_add=True)
 
 
 class DenonDevice(MediaPlayerEntity):
     """Representation of a Denon Media Player Device."""
 
-    def __init__(self, receiver, unique_id, config_entry):
+    def __init__(
+        self,
+        receiver: DenonAVR,
+        unique_id: str,
+        config_entry: config_entries.ConfigEntry,
+    ):
         """Initialize the device."""
         self._receiver = receiver
-        self._name = self._receiver.name
         self._unique_id = unique_id
         self._config_entry = config_entry
-        self._muted = self._receiver.muted
-        self._volume = self._receiver.volume
-        self._current_source = self._receiver.input_func
-        self._source_list = self._receiver.input_func_list
-        self._state = self._receiver.state
-        self._power = self._receiver.power
-        self._media_image_url = self._receiver.image_url
-        self._title = self._receiver.title
-        self._artist = self._receiver.artist
-        self._album = self._receiver.album
-        self._band = self._receiver.band
-        self._frequency = self._receiver.frequency
-        self._station = self._receiver.station
-
-        self._sound_mode_support = self._receiver.support_sound_mode
-        if self._sound_mode_support:
-            self._sound_mode = self._receiver.sound_mode
-            self._sound_mode_raw = self._receiver.sound_mode_raw
-            self._sound_mode_list = self._receiver.sound_mode_list
-        else:
-            self._sound_mode = None
-            self._sound_mode_raw = None
-            self._sound_mode_list = None
 
         self._supported_features_base = SUPPORT_DENON
         self._supported_features_base |= (
-            self._sound_mode_support and SUPPORT_SELECT_SOUND_MODE
+            self._receiver.support_sound_mode and SUPPORT_SELECT_SOUND_MODE
         )
+        self._available = True
+
+    def async_log_errors(  # pylint: disable=no-self-argument
+        func: Coroutine,
+    ) -> Coroutine:
+        """
+        Log errors occurred when calling a Denon AVR receiver.
+
+        Decorates methods of DenonDevice class.
+        Declaration of staticmethod for this method is at the end of this class.
+        """
+
+        @wraps(func)
+        async def wrapper(self, *args, **kwargs):
+            available = True
```

Comment:
Maybe disable # pylint: disable=protected-access at the top of this function if the checks will let you.

Reply:
this works